# M3DUtil/SampleCtrlNode __ct__18SampleCtrlMaterialFP11J3DMaterial: exact
# levers: name-call L110: name receiver material->getColorChan(i) J3DColorChan*
--- a/src/M3DUtil/SampleCtrlNode.cpp
+++ b/src/M3DUtil/SampleCtrlNode.cpp
@@ -107,7 +107,8 @@
 		unk18[i].mAmbSrc    = material->getColorChan(i)->getAmbSrc();
 		unk18[i].mLightMask = material->getColorChan(i)->getLightMask();
 		unk18[i].mDiffuseFn = material->getColorChan(i)->getDiffuseFn();
-		unk18[i].mAttnFn    = material->getColorChan(i)->getAttnFn();
+		J3DColorChan* colorChan6 = material->getColorChan(i);
+		unk18[i].mAttnFn    = colorChan6->getAttnFn();
 	}
 
 	unk38 = *(J3DTevOrderInfo*)material->getTevOrder(0);
